app/assets/javascripts/components/authentication/signup_form.es6.jsx

```diff
@@ -19,19 +31,51 @@ class SignupForm extends Component {
   createUser(event) {
     var params = {
       registration: {
-        email: 'warrenshen@berkeley.edu',
-        birthday: '2015-1-1',
-        first_name: 'Warren',
-        last_name: 'Shen',
-        password: 'password',
-        password_confirmation: 'password',
+        email: this.state.email,
+        birthday: this.state.birthday,
+        first_name: this.state.first_name,
+        last_name: this.state.last_name,
+        password: this.state.password,
+        password_confirmation: this.state.password_confirmation,
       },
     };
     var resolve = (response) => { console.log('Sign up success'); window.location = RouteConstants.students.index };
-    var reject = (response) => { console.log(response) };
+    var reject = (response) => { console.log(response); this.setState({errors: true}) };
     Requester.post(RouteConstants.users.create, params, resolve, reject);
   }
 
+  generateHandler(field) {
+    var state = {};
+    return function(event) {
+      state[field] = event.target.value;
+      this.setState(state);
+    }.bind(this);
```

Comment:
Let's use the rocket syntax now:
return (event) => {
      state[field] = event.target.value;
      this.setState(state);
 };
```